runhouse/servers/obj_store.py

```diff
@@ -1513,28 +1519,278 @@ def status_local(self):
+                for node_property in env_node_info
+                if ":" in node_property
+            }
+
+            node_ip = env_node_info.get("node_ip")
+            env_servlet_pid = int(env_actor_info.get("PID"))
+            if not cluster_config.get("resource_subtype") == "Cluster":
+                stable_internal_external_ips = cluster_config.get(
+                    "stable_internal_external_ips"
+                )
+                for ips_set in stable_internal_external_ips:
+                    internal_ip, external_ip = ips_set[0], ips_set[1]
+                    if internal_ip == node_ip:
+                        # head ip == cluster address == cluster.ips[0]
+                        if ips_set[1] == cluster_config.get("ips")[0]:
+                            node_name = f"head ({external_ip})"
+                        else:
+                            node_name = f"worker{stable_internal_external_ips.index(ips_set) + 1} ({external_ip}"
+            else:
+                # a case it is a BYO cluster, assume that first ip in the ips list is the head.
+                ips = cluster_config.get("ips")
+                if node_ip == ips[0]:
+                    node_name = f"head ({node_ip})"
+                else:
+                    node_name = f"worker{ips.index(node_ip) + 1} ({node_ip})"
+
+            try:
+                env_servlet_process = psutil.Process(pid=env_servlet_pid)
+                memory_size_bytes = env_servlet_process.memory_full_info().uss
+                cpu_usage_percent = env_servlet_process.cpu_percent(interval=1)
+                env_memory_usage = {
+                    "memory_size_bytes": memory_size_bytes,
+                    "cpu_usage_percent": cpu_usage_percent,
+                    "memory_percent_from_cluster": (memory_size_bytes / total_memory)
+                    * 100,
+                    "total_cluster_memory": total_memory,
+                    "env_memory_info": psutil.virtual_memory(),
+                }
+            except psutil.NoSuchProcess:
+                env_memory_usage = {}
+
+            return (
+                env_memory_usage,
+                node_name,
+                total_memory,
+                env_servlet_pid,
+                env_actor_info,
+                node_ip,
+            )
+
+    def _get_env_gpu_usage(self, env_servlet_pid):
+        import subprocess
+
+        try:
+            gpu_general_info = (
+                subprocess.run(
+                    [
+                        "nvidia-smi",
+                        "--query-gpu=utilization.gpu,memory.total,count",
+                        "--format=csv,noheader,nounits",
+                    ],
+                    stdout=subprocess.PIPE,
+                )
+                .stdout.decode("utf-8")
+                .strip()
+                .split(", ")
+            )
+            if gpu_general_info[0] == NO_GPU_NVIDIA_MSG:
```

Comment:
i think we can reuse existing functionality here - `_detect_cuda_version_or_cpu`